<launch>
    <arg name="paused" default="false" />
    <arg name="use_sim_time" default="true" />
    <arg name="gui" default="true" />
    <arg name="headless" default="false" />
    <arg name="debug" default="false" />
    <arg name="model" default="$(find quadrotor_sim)/urdf/quadrotor.xacro.urdf" />

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg paused)" />
        <arg name="use_sim_time" value="$(arg use_sim_time)" />
        <arg name="headless" value="$(arg headless)" />
    </include>

    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />

    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
        args="-urdf -model quadrotor -param robot_description"/>

    <arg name="node_start_delay" default="2.0" />  
    <node name="nonlinear_controller" pkg="quadrotor_sim" type="nonlinear_controller" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " />

</launch>